### ISSUE TITLE
Sqlite "Database is Locked" Error

### ISSUE THREAD

**Issue by agoops** [Hirevo/alexandrie#22]
In running some builds that reference many crates, against a Alexandrie+Sqlite setup, I started getting a 500 error from cargo, where Alexandrie logs showed:

```
SqlError("Database is locked")
```

To reproduce:
1) Stand up API using sqlite
2) Publish 4-5 crates
3) Have local project that depends on those 4-5 crates 
4) Repeatedly run build from clean environment (reset $CARGO_HOME each time)

It seems sqlite doesn't allow multiple writes at the same time.
To address it, I had to set sqlite to a WriteAheadLog mode:
```
sqlite3 alexandrie.db 'PRAGMA journal_mode=WAL;'
```
and limit the number of connections in the connection pool to 1

```rust
let mut builder = r2d2::Builder::default();
builder = builder.max_size(1);
```
(seemingly, because still only 1 connection is allowed to hold onto the corresponding `alexandrie.wal` file that gets created)

I modified `DatabaseConfig` in this [commit](https://github.com/agoops/alexandrie/commit/809fa4541850975580a227a78e5a507d8bfbe067) to accept an optional `connection_pool_max_size`. These 2 fixes, and restarting the app, solved the "Database is Locked" problem I was seeing.

Let me know what you think, or other suggestions to fix this or extend the Database configuration options to accommodate sqlite. I can open a PR for this if you think this is an appropriate change.

**Comment by Hirevo** [comment]
Hello,  
Thanks for reaching out about this, I am not a big user of SQLite, so I am not very familiar with how SQLite operates, and I made my tests with a local database only on a few endpoints with very light loads.  

I did some research and yeah, it seems to be because of concurrent writes and reads, indeed.  
When downloading crates, the writes comes from the download count increment in the database.  

I think the modifications you did to make the maximum number of open connections configurable is indeed needed.  
So, yes, I would happily accept a PR for this, let me know if you want to open it yourself or if I may replicate the changes you did.
